sed -e 's|^/\* \(#define LINUX\) \*/|\1|' \
    -e 's|^/\* \(#define TIMED_DELAY\) \*/|\1|' -i include/unixconf.h

sed -e '/^HACKDIR/ s|/games/lib/\$(GAME)dir|/var/games/nethack/|' \
    -e '/^SHELLDIR/ s|/games|/usr/bin|' \
    -e '/^VARDIRPERM/ s|0755|0775|' \
    -e '/^VARFILEPERM/ s|0600|0664|' \
    -e '/^GAMEPERM/ s|0755|02755|' \
    -e 's|\(DSYSCF_FILE=\)\\"[^"]*\\"|\1\\"/var/games/nethack/sysconf\\"|' \
    -e 's|\(DHACKDIR=\)\\"[^"]*\\"|\1\\"/etc/nethack/\\"|' -i sys/unix/hints/linux.500

sed -e 's|^#GAMEUID.*|GAMEUID = root|' \
    -e 's|^#GAMEGRP.*|GAMEGRP = games|' \
    -e '/^FILEPERM\s*=/ s|0644|0664|' \
    -e '/^DIRPERM\s*=/ s|0755|0775|' -i sys/unix/Makefile.top

sed -e "/^MANDIR\s*=/s|/usr/man/man6|$pkgdir/usr/share/man/man6|" -i sys/unix/Makefile.doc

diff --git a/include/tradstdc.h b/include/tradstdc.h
index a202c0c7b..4b20ec330 100644
--- a/include/tradstdc.h
+++ b/include/tradstdc.h
@@ -444,7 +444,7 @@ typedef genericptr genericptr_t; /* (void *) or (char *) */
 #if (!defined(__linux__) && !defined(MACOS)) || defined(GCC_URWARN)
 /* disable gcc's __attribute__((__warn_unused_result__)) since explicitly
    discarding the result by casting to (void) is not accepted as a 'use' */
-#define __warn_unused_result__ /*empty*/
+#define __warn_unused_result__ __unused__
 #define warn_unused_result /*empty*/
 #endif  /* GCC_URWARN || !__linux || !MACOS */
 #endif  /* __GNUC__ || clang masquerading as __GNUC__==5 */
diff --git a/include/unixconf.h b/include/unixconf.h
index e7ee91565..42f4f6de3 100644
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -112,7 +112,7 @@
  * "extra output" method is used, but not all systems provide access to
  * a fine-grained timer.
  */
-/* #define TIMED_DELAY */ /* usleep() */
+#define TIMED_DELAY /* usleep() */
 #endif
 #if defined(MACOS) && !defined(TIMED_DELAY)
 #define TIMED_DELAY
diff --git a/sys/unix/Makefile.top b/sys/unix/Makefile.top
index 5ecca3232..dd4b89520 100644
--- a/sys/unix/Makefile.top
+++ b/sys/unix/Makefile.top
@@ -25,16 +25,16 @@ NHSROOT=.
 GAME     = nethack
 # GAME     = nethack.exe
 # GAME     = nethack.prg
-#GAMEUID  = games
-#GAMEGRP  = bin
+GAMEUID = root
+GAMEGRP = games
 
 #	Permissions - some places use setgid instead of setuid, for instance.
 #	See also the option "SECURE" in include/config.h.
 #GAMEPERM = 04755
-FILEPERM = 0644
+FILEPERM = 0664
 # VARFILEPERM = 0644
 EXEPERM  = 0755
-DIRPERM  = 0755
+DIRPERM  = 0775
 # VARDIRPERM = 0755
 
 #	VARDIR may also appear in unixconf.h as "VAR_PLAYGROUND" else HACKDIR
diff --git a/sys/unix/hints/linux.500 b/sys/unix/hints/linux.500
index b822b6e50..9b34b732b 100755
--- a/sys/unix/hints/linux.500
+++ b/sys/unix/hints/linux.500
@@ -158,10 +158,9 @@ endif      #USE_CURSESLIB
 
 # NetHack sources control
 NHCFLAGS+=-DDLB
-NHCFLAGS+=-DHACKDIR=\"$(HACKDIR)\"
+NHCFLAGS+=-DHACKDIR=\"/var/games/nethack/\"
 NHCFLAGS+=-DDEFAULT_WINDOW_SYS=\"$(WANT_DEFAULT)\"
-NHCFLAGS+=-DSYSCF -DSYSCF_FILE=\"$(HACKDIR)/sysconf\" -DSECURE
-NHCFLAGS+=-DTIMED_DELAY
+NHCFLAGS+=-DSYSCF -DSYSCF_FILE=\"/etc/nethack/sysconf\" -DSECURE
 NHCFLAGS+=-DDUMPLOG
 NHCFLAGS+=-DCONFIG_ERROR_SECURE=FALSE
 #NHCFLAGS+=-DGREPPATH=\"/usr/bin/grep\"
@@ -412,18 +411,18 @@ PREFIX=$(abspath $(NHSROOT))
 #SHELLDIR=
 HACKDIR=$(PREFIX)/playground
 GAMEPERM = 0700
-VARFILEPERM = 0600
+VARFILEPERM = 0664
 VARDIRPERM = 0700
-CFLAGS+=-DSYSCF -DSYSCF_FILE=\"$(HACKDIR)/sysconf\" -DSECURE
+CFLAGS+=-DSYSCF -DSYSCF_FILE=\"/etc/nethack/sysconf\" -DSECURE
 MOREALL=$(MAKE) install
 else #!WANT_SOURCE_INSTALL
 #PREFIX=/usr
 PREFIX=$(wildcard ~)/nh/install
 HACKDIR=$(PREFIX)/games/lib/nethackdir
-SHELLDIR = $(PREFIX)/games
-VARDIRPERM = 0755
-VARFILEPERM = 0600
-GAMEPERM = 0755
+SHELLDIR = $(PREFIX)/usr/bin
+VARDIRPERM = 0775
+VARFILEPERM = 0664
+GAMEPERM = 02755
 endif #?WANT_SOURCE_INSTALL
 
 INSTDIR=$(HACKDIR)
